render/auto.go

```diff
@@ -113,34 +114,38 @@ func (ir htmlAutoRenderer) Render(w io.Writer, data Data) error {
 				return err
 			}
 			if data["method"] == "PUT" {
-				return ir.HTML(fmt.Sprintf("%s/edit.html", templatePrefix)).Render(w, data)
+				return ir.osIndependentHTML(fmt.Sprintf("%s/edit.html", templatePrefix)).Render(w, data)
 			}
-			return ir.HTML(fmt.Sprintf("%s/new.html", templatePrefix)).Render(w, data)
+			return ir.osIndependentHTML(fmt.Sprintf("%s/new.html", templatePrefix)).Render(w, data)
 		}
 		return nil
 	}
 	cp, ok := data["current_path"].(string)
 
 	defCase := func() error {
-		return ir.HTML(fmt.Sprintf("%s/%s.html", templatePrefix, "index")).Render(w, data)
+		return ir.osIndependentHTML(fmt.Sprintf("%s/%s.html", templatePrefix, "index")).Render(w, data)
 	}
 	if !ok {
 		return defCase()
 	}
 
 	if strings.HasSuffix(cp, "/edit/") {
-		return ir.HTML(fmt.Sprintf("%s/edit.html", templatePrefix)).Render(w, data)
+		return ir.osIndependentHTML(fmt.Sprintf("%s/edit.html", templatePrefix)).Render(w, data)
 	}
 	if strings.HasSuffix(cp, "/new/") {
-		return ir.HTML(fmt.Sprintf("%s/new.html", templatePrefix)).Render(w, data)
+		return ir.osIndependentHTML(fmt.Sprintf("%s/new.html", templatePrefix)).Render(w, data)
 	}
 
 	if !isPlural {
-		return ir.HTML(fmt.Sprintf("%s/show.html", templatePrefix)).Render(w, data)
+		return ir.osIndependentHTML(fmt.Sprintf("%s/show.html", templatePrefix)).Render(w, data)
 	}
 	return defCase()
 }
 
+func (ir htmlAutoRenderer) osIndependentHTML(path string) Renderer {
```

Comment:
If we use `filepath.Join` we will not need this method.

Reply:
I think it is a good idea. Just KISS. :-)